core/STFtrace/STFLogger.cpp

```diff
@@ -0,0 +1,53 @@
+
+#include "STFLogger.hpp"
+#include "arch/RegisterSet.hpp"
+#include "core/AtlasState.hpp"
+
+namespace atlas{
+// CONSTRUCTORS 
+    STFLogger::STFLogger(){
+        stf_writer_ = new stf::STFWriter();
+    } 
+
+// DESTRUCTORS
+    STFLogger::~STFLogger(){
+        if(enable)  stf_writer_->close();
+        delete stf_writer_;
+        stf_writer_ = nullptr;
+        enable = false;
+    }
+
+//METHODS
+    //stf_enable -> user can enable and disable STF Trace Generation
+    //width -> register width (32 or 64)
+    //pc -> initial program counter
+    void STFLogger::initialize(const bool stf_enable, const uint32_t width, uint64_t pc){
```

Comment:
The use of an `initialize` method is not recommended and makes class design difficult.  Specifically, you are requiring the user of your class to _always_ do this:
```
   Foo f;
   f.initialize(var1, var2, var3);
```
which requires all of the member variables in `Foo` to have default construction and/or copy-able/assign-able behavior.

This is more efficient and does not require default construction for member variables:
```
    Foo f(var1, var2, var3);
```
So my suggestion here is -- remove the `initialize` method and create a constructor that takes these arguments.